src/app/log-tool/export-modal/export-modal.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ModalDirective } from 'ngx-bootstrap';
+import { Subscription } from 'rxjs';
+import { WindowRefService } from '../../indexedDb/window-ref.service';
+import { LogToolDbService } from '../log-tool-db.service';
+import { LogToolDbData } from '../log-tool-models';
+import { LogToolService } from '../log-tool.service';
+
+@Component({
+  selector: 'app-export-modal',
+  templateUrl: './export-modal.component.html',
+  styleUrls: ['./export-modal.component.css']
+})
+export class ExportModalComponent implements OnInit {
+  
+  @ViewChild('exportModal', { static: false }) public exportModal: ModalDirective;
+  
+  dataExists: boolean = false;
+  dataAvailableSub: Subscription;
+  dataAvailable: Date;
+  data: LogToolDbData;
+  exportName: string;
+  constructor(private logToolService: LogToolService, private logToolDbService: LogToolDbService, private windowRefService: WindowRefService) { }
+
+  ngOnInit(){
+  }
+
+  ngAfterViewInit() {
+    this.showExportModal();
+  }
+
+  showExportModal() {
+    this.exportModal.show();
+  }
+
+  closeExportData(){
+    this.exportModal.hide()
+    this.exportModal.onHidden.subscribe(val => {
+      this.logToolService.openExportData.next(false);
+    });
+  }
+  ngOnDestroy() {
+    if (this.dataAvailableSub) {
+      this.dataAvailableSub.unsubscribe();
+    }
+  }
+
+  getExportData(){
+    this.dataAvailableSub = this.logToolDbService.previousDataAvailable.subscribe(val => {
```

Comment:
Looks like your using dataAvailable to show last modified date. The object LogToolDb has a propery "modifiedDate" you can use that on the "data" object and remove this subscription logic